occurrences/admin.py

```diff
@@ -34,23 +34,51 @@ class OccurrenceInline(admin.TabularInline):
     model = Occurrence.study_groups.through
     extra = 0
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    # Without setting these add and delete permissions to false, the page loads
+    # slowly due to a large number of SQL queries
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
 
 class EventQueueEnrolmentInline(admin.TabularInline):
     model = EventQueueEnrolment
     extra = 0
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    # Without setting these add and delete permissions to false, the page loads
+    # slowly due to a large number of SQL queries
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
```

Comment:
The enrolments are something that we really might have to edit from the Django admin.

```suggestion
    readonly_fields = ['person_deleted_at']
    autocomplete_fields = ['person', 'p_event']
```